backend/protocol_rpc/endpoints.py

```diff
@@ -531,6 +536,113 @@ def set_finality_window_time(consensus: ConsensusAlgorithm, time: int) -> None:
     consensus.set_finality_window_time(time)
 
 
+def get_chain_id() -> str:
+    return hex(61_999)
+
+
+def get_net_version() -> str:
+    return "61999"
+
+
+def get_block_number(transactions_processor: TransactionsProcessor) -> str:
+    transaction_count = transactions_processor.get_highest_timestamp()
+    return hex(transaction_count)
+
+
+def get_block_by_number(
+    transactions_processor: TransactionsProcessor, block_number: str, full_tx: bool
+) -> dict | None:
+    try:
+        block_number_int = int(block_number, 16)
+    except ValueError:
+        raise JSONRPCError(f"Invalid block number format: {block_number}")
+
+    block_details = transactions_processor.get_transactions_for_block(
+        block_number_int, include_full_tx=full_tx
+    )
+
+    if not block_details:
+        raise JSONRPCError(f"Block not found for number: {block_number}")
+
+    return block_details
```

Comment:
Why is the signature `dict | None`? I don't see anywhere where it could return `None`

Reply:
Fixed